runorders.cpp

```diff
@@ -531,6 +531,25 @@ void Game::RunDestroyOrders()
 	}
 }
 
+int get_structure_health(const Object *o) {
+	int cost = ObjectDefs[o->type].cost;
+	int health = cost - o->incomplete;
+
+	return health;
+}
+
+int get_structure_points(const Object *o, int min_destroy, int max_destroy_pct) {
+	// total cost or maxium HP of the building
+	int cost = ObjectDefs[o->type].cost;
+	int health = get_structure_health(o);
+
+	// points that can be destroyed, they are reduced by already destroyed points this turn
+	int points = std::max(0, std::max(min_destroy, cost * max_destroy_pct / 100) - o->destroyed);
+
+	// amount avaiable for destruction, but no more than there are HP of the building itself
+	return std::min(health, points);
+}
+
```

Comment:
see comment above.. if these are meant to be usable generally, make them part of Object rather than passing it in.